WD-Firmware/examples/build/SConstruct

```diff
@@ -14,65 +14,43 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 '''
+
+#### scons custom arguments ####
+tagretName = ARGUMENTS.get('target', '')
+
+#downloadFlag = ARGUMENTS.get('download', '0')
+
 # set imports and variables of sconstruct
 import sys
 import os
 import time
-import platform
+import utils
+from demos.factory import generate
+
+# generate the demo factory and set the demo 
+objFactory = generate()
+objDemo = objFactory.setDemo()
+
+# set the output files for the demo
+objDemo.set_outputfiles_names(prefix=tagretName)
 
 root_dir = os.path.join(os.getcwd(),"..", "..")
- 
-# this function creates a header string
-def header_create(str_arr):
-   header = '\n _______________________________________\n'
-   for line in str_arr:
-      header += '\n\t' + line
-   header += '\n _______________________________________\n'
-   return header
-
-# add 'size' util output to the map file
-def produce_sections_size(target, source, env):
-   global map_file_name
-   text = 'Sections size summary'
-   # check if there was any change in the .map file
-   if not text in open(map_file_name).read():
-      # .map was updated so add the sections size
-      elf_name = str(source[0])
-      size_util_name = os.path.join(Env['TOOLS_BASE_DIR'], 'riscv64-unknown-elf-size')
-      headere = header_create([text])
-      os.system('echo "' + headere + '" >> ' + map_file_name)
-      os.system(size_util_name + ' ' + elf_name + ' >> ' + map_file_name) 
-   return None
+
+listSupportedBoards = ['hifive1', 'hifive-un', 'nexys_a7_swerv1']
+dependency_list = ['libftdi1-2',]

```

Comment:
add comment (what is the dependency list)

Reply:
1. also add comment for listSupported...
2. convention for all